python/lsst/ap/association/association.py

```diff
@@ -494,11 +513,49 @@ def update_dia_objects(self, dia_objects, updated_obj_ids, exposure, ppdb):
             #     Define and improve flagging for DiaObjects
             # Set a flag on this DiaObject if any DiaSource that makes up this
             # object has a flag set for any reason.
-            if np.any(obj_dia_sources.get("flags") > 0):
-                dia_object.set("flags", 1)
+            if np.any(obj_dia_sources["flags"] > 0):
+                updated_dia_object["flags"] = 1
+            else:
+                updated_dia_object["flags"] = 0
+
+            updated_dia_objects.append(updated_dia_object)
+
+        # Initialize DataFrame with full set of columns to properly set
+        # NaN/Null values.
+        updated_dia_objects = pandas.DataFrame(
+            data=updated_dia_objects,
+            columns=list(ppdb._schema.getColumnMap('DiaObject').keys()))
 
         ppdb.storeDiaObjects(updated_dia_objects, dateTime.toPython())
 
+        updated_dia_objects.set_index(["diaObjectId"], inplace=True)
+
+        return dia_objects[
+            np.logical_not(dia_object_used.loc[:, "used"])].append(updated_dia_objects,
+                                                                   sort=True)
+
+    def _initialize_dia_object(self, obj_id):
+        """Input default values for non-Nullable DiaObject columns.
+
+        Parameters
+        ----------
+        obj_id : `int`
+            Id to asign to the new object.
+
+        Returns
+        -------
+        new_dia_object : `dict`
+            Dictionary with default values.
+        """
+        new_dia_object = {"diaObjectId": obj_id,
+                          "pmParallaxNdata": 0,
+                          "nearbyObj1": 0,
+                          "nearbyObj2": 0,
+                          "nearbyObj3": 0}
+        for f in ["u", "g", "r", "i", "z", "y"]:
```

Comment:
Filter names are not currently configurable as the columns in the Ppdb are fixed elsewhere. The best place to do this I think would be somewhere in the ppdb so that the columns of the DB match up with those here.